code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -135,10 +135,11 @@
 	. = 1
 
 /datum/reagent/drug/krokodil/addiction_act_stage4(mob/living/carbon/human/M)
-	if(M.has_dna() && M.dna.species.id != "zombie")
+	CHECK_DNA_AND_SPECIES(M)
+	if(istype(M.dna.species, /datum/species/krokodil_addict))
```

Comment:
because species id sets what sprites you use, and isn't guaranteed unique. For example, the krokodil_addict uses the zombie id, but still is human in every other regard.